src/test/scala/nl/knaw/dans/lib/string/StringExtensionsSpec.scala

```diff
@@ -93,4 +95,28 @@ class StringExtensionsSpec extends FlatSpec with Matchers with OptionValues {
     val input = "ab c "
     input.emptyIfBlank shouldBe input
   }
+
+  "toUUID" should "return a UUID on a valid input" in {
+    val input = "12341234-1234-1234-1234-123412341234"
+    val expectedOutput = UUID.fromString(input)
+    
+    input.toUUID.right.value shouldBe expectedOutput
+  }
+  
+  it should "return an error on an invalid input" in {
+    val input = "xyz"
+    
+    input.toUUID.left.value shouldBe UUIDError(input)
+  }
+
+  it should "return an error on an input that is valid according to Java's UUID parser, but does not conform to the pattern of a UUID" in {
+    val input = "1-2-3-4-5"
+    val javaOutput = UUID.fromString(input)
+
+    // this is what the Java parser does, but this is not what we want!
+    javaOutput.toString shouldBe "00000001-0002-0003-0004-000000000005"
```

Comment:
Do we need to check that the Java parser does this? Will this fail in current Java versions? (You mentioned that the problem has been fixed there.)

Reply:
That's a different bug you're talking about...
https://bugs.openjdk.java.net/browse/JDK-8159339